include/tdi/common/tdi_defs.h

```diff
@@ -109,6 +112,7 @@ enum tdi_field_data_type_e {
   TDI_FIELD_DATA_TYPE_STRING_ARR = 8,
   TDI_FIELD_DATA_TYPE_UNKNOWN = 9,
 };
+typedef enum tdi_field_data_type_e tdi_field_data_type_e;
```

Comment:
```suggestion
typedef enum {
  TDI_FIELD_DATA_TYPE_INT_ARR = 0,
  TDI_FIELD_DATA_TYPE_BOOL_ARR = 1,
  TDI_FIELD_DATA_TYPE_UINT64 = 2,
  TDI_FIELD_DATA_TYPE_BYTE_STREAM = 3,
  TDI_FIELD_DATA_TYPE_FLOAT = 4,
  TDI_FIELD_DATA_TYPE_CONTAINER = 5,
  TDI_FIELD_DATA_TYPE_STRING = 6,
  TDI_FIELD_DATA_TYPE_BOOL = 7,
  TDI_FIELD_DATA_TYPE_STRING_ARR = 8,
  TDI_FIELD_DATA_TYPE_UNKNOWN = 9,
} tdi_field_data_type_e;
```

Why not do it in one line itself rather than repeating it another line and risking a typo ?

Reply:
I am using the second method which compatible with C++.  In this way, we can also pass with enum tdi_field_data_type_e ...
https://riptutorial.com/c/example/6564/typedef-enum